src/platform/packages/shared/kbn-scout/src/common/services/custom_role.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { EsClient, KbnClient } from '.';
+
+export interface KibanaRole {
+  elasticsearch: {
+    cluster: string[];
+    indices: Array<{
+      names: string[];
+      privileges: string[];
+      allow_restricted_indices?: boolean | undefined;
+    }>;
+  };
+  kibana: Array<{
+    base: string[];
+    feature: Record<string, string[]>;
+    spaces: string[];
+  }>;
+}
+
+export interface ElasticsearchRoleDescriptor {
+  cluster?: string[];
+  indices?: Array<{
+    names: string[];
+    privileges: string[];
+    allow_restricted_indices?: boolean;
+  }>;
+  applications?: Array<{
+    application: string;
+    privileges: string[];
+    resources: string[];
+  }>;
+  run_as?: string[];
+}
+
+export const createCustomRole = async (
+  kbnClient: KbnClient,
+  customRoleName: string,
+  role: KibanaRole
+) => {
+  const { status } = await kbnClient.request({
+    method: 'PUT',
+    path: `/api/security/role/${customRoleName}`,
+    body: role,
+  });
```

Comment:
It looks like we don't have the equivalent of Elasticsearch's client `client.security.putRole` for the `kbnClient`, which is fine.

In which situations in makes more sense to add new functionality to the `kbnClient` vs sending the request directly using `kbnClient.request`?

Reply:
>In which situations it makes more sense to add new functionality to the kbnClient vs sending the request directly using kbnClient.request?

[Elasticsearch js client](http://npmjs.com/package/@elastic/elasticsearch) is a published library with a dedicated team, kbnClient is something we use internally and currently I don't expect much work around it.